api/lib/application/saml/saml-controller.js

```diff
@@ -23,6 +24,12 @@ module.exports = {
     }
 
     try {
+      if (features.garAccessV2) {
+        const studentReconciliationToken = tokenService.createTokenForStudentReconciliation({ userAttributes });
+
+        return h.redirect(`/campagnes?student=${studentReconciliationToken}`);
+      }
```

Comment:
Vu que le redirect est un early-exit, qu'il n'y a qu'une alternative, et que l'algo est similaire, je propose de tout rapatrier dans le else
```suggestion
      if (features.garAccessV2) {
        const studentReconciliationToken = tokenService.createTokenForStudentReconciliation({ userAttributes });
        return h.redirect(`/campagnes?student=${studentReconciliationToken}`);
      } else {
            const user = await usecases.getOrCreateSamlUser({ userAttributes });
           const token = tokenService.createTokenFromUser(user, 'external');
           return h.redirect(`/?token=${encodeURIComponent(token)}&user-id=${user.id}`);
      }
```

Reply:
C'est justement pour toutes les raisons que tu as cité que le `else` n'est pas nécessaire ici.